trace/opentracing.go

```diff
@@ -409,11 +414,19 @@ func (t Tracer) StartSpan(operationName string, opts ...opentracing.StartSpanOpt
 
 	for k, v := range sso.Tags {
 		span.SetTag(k, v)
-		if k == "name" {
+		if k == NameKey {
 			span.Name = v.(string)
 		}
 	}
 
+	if span.Name == "" {
```

Comment:
Hrm, this doesn't need fixing now, but it feels like this should be two different functions (: